### PR TITLE
feat: Shopping lists API

### PR DESCRIPTION
**Pull request by JoseCapela**
Created Shopping lists API services.

## Description
Added the following operations:
Get Shoppinglists
Get Shoppinglist By Id
Post Shoppinglist
Put Shoppinglist
Delete Shoppinglist

## Motivation and Context
It adds the Shopping Lists feature

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation improvement/changes

## Checklist:
- [x] My code passes the code style as defined by TSLint.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes and the overall coverage did not decrease.
- [x] I ran all tests locally and they pass.
- [x] All unit tests pass on CircleCi.

### CODE REVIEW

File: cif/shoppinglists/src/actions/shoppinglists.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    .setClient(PatchShoppingListClient)
+    .setErrorType(ERROR_TYPE)
+    .activate();
+}
+export const patch = patchShoppingList;
+
+function postShoppingList(input: Input): Promise<Input> {
+  return new SequenceAction(input)
+    .setClient(PostShoppingListClient)
+    .setMapper(ShoppingListCreationMapperMapper)
```
Comment (cduque89) [inline cif/shoppinglists/src/actions/shoppinglists.ts]:
```suggestion
    .setMapper(ShoppingListCreationMapper)
```

File: cif/shoppinglists/src/clients/PostShoppingListClient.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
+/*
```
Comment (cduque89) [inline cif/shoppinglists/src/clients/PostShoppingListClient.ts]: Maybe we can use the same client as for PostCart

File: cif/shoppinglists/src/mappers/PagedResponseShoppingListMapper.ts [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    const {
+      carts = [],
+    } = dto;
+
+    const {
+      limit = 20,
+      offset = 0,
+    } = this.actionParameters;
+
+    const totalResults = carts.length;   // TODO: como saber o total real?  fazer outra chamada com "fields = BASIC" e sem limites para saber?
```
Comment (cduque89) [inline cif/shoppinglists/src/mappers/PagedResponseShoppingListMapper.ts]: Review TODO

File: cif/shoppinglists/src/mappers/ShoppingListEntryMapper.ts [hunk 4]
```diff
@@ -0,0 +1,92 @@
+/*
```
Comment (cduque89) [inline cif/shoppinglists/src/mappers/ShoppingListEntryMapper.ts]: Review if it's possible to use CartEntryMapper

Comment (JoseCapela) [reply cif/shoppinglists/src/mappers/ShoppingListEntryMapper.ts]: Although they are similar, there are a few differences between the CartEntry and ShoppingListEntry data structures (name of same fields, the location where the price components are used, etc) so I think it's best to keep them in separate files

**Pull request merged**